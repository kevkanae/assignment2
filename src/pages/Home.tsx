import {
  Box,
  Button,
  Grid,
  TextField,
  MenuItem,
  Typography,
} from "@mui/material";
import { useEffect, useState } from "react";
import { useNavigate } from "react-router-dom";
import { languages } from "../constants/Languages";

const Home = () => {
  const navigate = useNavigate();
  const [name, setName] = useState<string>("");
  const [lang, setLang] = useState<string>("JS");

  const handleName = (e: React.ChangeEvent<HTMLInputElement>) =>
    setName(e.currentTarget.value);
  const handleLang = (e: React.ChangeEvent<HTMLInputElement>) =>
    setLang(e.target.value);

  const handleSubmit = () => {
    if (name !== "" && lang !== "") {
      console.log({ name, lang });
      localStorage.setItem("User", JSON.stringify({ name, lang }));
      navigate("/quiz");
    }
  };

  useEffect(() => {
    if (Object.keys(localStorage).length !== 0) {
      localStorage.clear();
    }
  }, []);

  return (
    <>
      <Box
        sx={{
          display: "flex",
          flexDirection: "column",
          alignItems: "center",
          justifyContent: "center",
          height: "100vh",
          width: "100vw",
          backgroundColor: "#E5E5E5",
        }}
      >
        <Box
          sx={{
            display: "flex",
            flexDirection: "column",
            alignItems: "center",
            width: "56vw",
            backgroundColor: "#fff",
            borderRadius: 3,
            p: "1rem",
            color: "#000",
          }}
        >
          <Typography>Enter Your Details</Typography>
          <Grid
            container
            rowSpacing={2}
            columnSpacing={2}
            direction="row"
            justifyContent="space-evenly"
            alignItems="center"
          >
            {[
              <TextField
                label="Name"
                variant="outlined"
                fullWidth
                required
                role="Name-inp"
                onChange={handleName}
                value={name}
              />,
              <TextField
                id="outlined-select-currency"
                select
                label="Language"
                aria-label="Language"
                value={lang}
                onChange={handleLang}
              >
                {languages.map((option) => (
                  <MenuItem key={option.value} value={option.value}>
                    {option.label}
                  </MenuItem>
                ))}
              </TextField>,
              <TextField label="Age" variant="outlined" fullWidth />,
              <TextField label="Gender" variant="outlined" fullWidth />,
            ].map((x, i) => (
              <Grid key={i} item xs={6}>
                <Box sx={{ p: 1 }}>{x}</Box>
              </Grid>
            ))}
          </Grid>
          <Button role="sub-button" onClick={handleSubmit}>
            Submit
          </Button>
        </Box>
      </Box>
    </>
  );
};

export default Home;
