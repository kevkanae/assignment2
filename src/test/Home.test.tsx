import React from "react";
import renderer from "react-test-renderer";
import "@testing-library/jest-dom/extend-expect";
import { screen, render } from "@testing-library/react";
import Home from "../pages/Home";
import { BrowserRouter } from "react-router-dom";

test("Snapshot of Home component", () => {
  const comp = renderer.create(
    <BrowserRouter>
      <Home lang={""} name={"K"} setName={() => {}} setLang={() => {}} />
    </BrowserRouter>
  );
  let tree = comp.toJSON();
  expect(tree).toMatchSnapshot();
});

describe("In Home,", () => {
  test("is the header present", () => {
    render(
      <BrowserRouter>
        <Home lang={""} name={"K"} setName={() => {}} setLang={() => {}} />
      </BrowserRouter>
    );
    const heading = screen.getByText(/Enter Your Details/i);
    expect(heading).toBeInTheDocument();
  });

  test("is the name input present", () => {
    render(
      <BrowserRouter>
        <Home lang={""} name={"K"} setName={() => {}} setLang={() => {}} />
      </BrowserRouter>
    );
    const nameInput = screen.getByLabelText(/Name/i);
    expect(nameInput).toBeInTheDocument();
  });

  test("is the language input present", () => {
    render(
      <BrowserRouter>
        <Home lang={""} name={"K"} setName={() => {}} setLang={() => {}} />
      </BrowserRouter>
    );
    const languageInput = screen.getByLabelText(/Language-Select/i);
    expect(languageInput).toBeInTheDocument();
  });

  test("is the submit button present", () => {
    render(
      <BrowserRouter>
        <Home lang={""} name={"K"} setName={() => {}} setLang={() => {}} />
      </BrowserRouter>
    );
    const submitButton = screen.getByRole(/sub-button/i);
    expect(submitButton).toBeInTheDocument();
  });

  test("is the age input present", () => {
    render(
      <BrowserRouter>
        <Home lang={""} name={"K"} setName={() => {}} setLang={() => {}} />
      </BrowserRouter>
    );
    const age = screen.getByLabelText(/Enter Age/i);
    expect(age).toBeInTheDocument();
  });

  test("is the gender input present", () => {
    render(
      <BrowserRouter>
        <Home lang={""} name={"K"} setName={() => {}} setLang={() => {}} />
      </BrowserRouter>
    );
    const gender = screen.getByLabelText(/Gender/i);
    expect(gender).toBeInTheDocument();
  });
});
