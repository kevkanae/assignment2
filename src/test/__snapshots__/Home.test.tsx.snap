// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot of Home component 1`] = `
<div
  className="MuiBox-root css-17ku038"
>
  <div
    className="MuiBox-root css-9uz7ai"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      Enter Your Details
    </p>
    <div
      className="MuiGrid-root MuiGrid-container css-l3t121-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
      >
        <div
          className="MuiBox-root css-hpgf8j"
        >
          <div
            className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            role="Name-inp"
          >
            <label
              className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink={false}
              htmlFor=":r0:"
              id=":r0:-label"
            >
              Name
              <span
                aria-hidden={true}
                className="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                disabled={false}
                id=":r0:"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={true}
                type="text"
                value=""
              />
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-1ftyaf0"
                >
                  <span>
                    Name
                     
                    *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
      >
        <div
          className="MuiBox-root css-hpgf8j"
        >
          <div
            aria-label="Language"
            className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          >
            <label
              className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled Mui-required css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink={true}
              htmlFor="outlined-select-currency"
              id="outlined-select-currency-label"
            >
              Language-Select
              <span
                aria-hidden={true}
                className="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              onClick={[Function]}
              variant="outlined"
            >
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="outlined-select-currency-label outlined-select-currency"
                className="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                id="outlined-select-currency"
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                role="button"
                tabIndex={0}
              >
                English
              </div>
              <input
                aria-hidden={true}
                autoFocus={false}
                className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                disabled={false}
                onAnimationStart={[Function]}
                onChange={[Function]}
                required={true}
                tabIndex={-1}
                value="ENG"
              />
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-1in441m"
                >
                  <span>
                    Language-Select
                     
                    *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
      >
        <div
          className="MuiBox-root css-hpgf8j"
        >
          <div
            className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          >
            <label
              className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink={false}
              htmlFor=":r2:"
              id=":r2:-label"
            >
              Enter Age
            </label>
            <div
              className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                disabled={false}
                id=":r2:"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="text"
              />
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-1ftyaf0"
                >
                  <span>
                    Enter Age
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
      >
        <div
          className="MuiBox-root css-hpgf8j"
        >
          <div
            className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          >
            <label
              className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink={false}
              htmlFor=":r3:"
              id=":r3:-label"
            >
              Enter Gender
            </label>
            <div
              className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                disabled={false}
                id=":r3:"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="text"
              />
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-1ftyaf0"
                >
                  <span>
                    Enter Gender
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
    </div>
    <button
      className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="sub-button"
      tabIndex={0}
      type="button"
    >
      Submit
    </button>
  </div>
</div>
`;
